dbg.h

```diff
@@ -73,24 +73,63 @@ namespace pretty_function {
 #if defined(__clang__)
 #define DBG_MACRO_PRETTY_FUNCTION __PRETTY_FUNCTION__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char *dbg_macro::type_name_impl() [T = ") - 1;
+    sizeof("const char *dbg::type_name_impl() [T = ") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof("]") - 1;
 #elif defined(__GNUC__) && !defined(__clang__)
 #define DBG_MACRO_PRETTY_FUNCTION __PRETTY_FUNCTION__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char* dbg_macro::type_name_impl() [with T = ") - 1;
+    sizeof("const char* dbg::type_name_impl() [with T = ") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof("]") - 1;
 #elif defined(_MSC_VER)
 #define DBG_MACRO_PRETTY_FUNCTION __FUNCSIG__
 static constexpr size_t PREFIX_LENGTH =
-    sizeof("const char *__cdecl dbg_macro::type_name_impl<") - 1;
+    sizeof("const char *__cdecl dbg::type_name_impl<") - 1;
 static constexpr size_t SUFFIX_LENGTH = sizeof(">(void)") - 1;
 #else
 #error "This compiler is currently not supported by dbg_macro."
 #endif
 
 }  // namespace pretty_function
 
+// Formatting helpers
+
+template <typename T>
+struct print_formatted {
+  static_assert(std::is_integral<T>::value && std::is_unsigned<T>::value,
```

Comment:
I learned a thing, very nice!

Reply:
You constraint values to be unsigned. That is a good things, but I think there is a lot of use-cases where we would like to print signed number as hexadecimal. Example:

![image](https://user-images.githubusercontent.com/5982507/71470578-e5c62080-27cc-11ea-8d4a-84d0968978da.png)

The prototype of this function specify `flags` to be a `signed int`. This is the first use-case among many others that came to my mind. Here, we would need to cast it to `unsigned int` before printing!

It is a matter of choice from you: do you want fast and easy debugging (we allow printing signed numbers as hexadecimal/octal) or do you want to force the user to be syntactically correct (force to cast to unsigned type)?

Reply:
> This is the first use-case among many others that came to my mind. Here, we would need to cast it to `unsigned int` before printing!

I was too lazy to implement the necessary logic :smile:. Added it now.

> It is a matter of choice from you: do you want fast and easy debugging (we allow printing signed numbers as hexadecimal/octal) or do you want to force the user to be syntactically correct (force to cast to unsigned type)?

I don't think it's a matter of choice. Printing a signed integral type as `-0xABC123` is definitely useful and doesn't require anyone to cast explicitly.